preserve_data.rb

```diff
@@ -0,0 +1,16 @@
+require 'json'
+
+def write_file(obj, filename)
+  data = obj.map(&:to_h)
+  data_json = JSON.pretty_generate(data)
+  File.write(filename, data_json, mode: 'w')
+end
+
+def read_file(filename)
+  if File.exist?(filename) and File.size(filename) != 0
+    data = File.read(filename)
+    JSON.parse(data)
+  else
+    []
+  end
+end
```

Comment:
- [ ] Hello Team! 👏 Fantastic job on implementing the methods for saving and reading data from JSON. 📝📖 While the functionality is in place, there's a small hiccup. According to the project guidelines, data should be saved on app exit, but the current implementation saves data immediately upon creation, deviating from the required behavior. 🚨 To address this, let's create an exit_program method. Inside this method, call the data-saving method to ensure it happens only during app exit. 🔄 Make this adjustment for a more aligned implementation with the project specifications. If you have any questions or need further clarification, feel free to reach out! 🌟


![Screenshot from 2023-11-15 18-33-20](https://github.com/tsheporamantso/opp_school_library/assets/112261237/da48569b-ee45-4d0e-8cee-ea61edfb5316)